spec.bs

```diff
@@ -743,7 +743,76 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
 <div algorithm>
   The <dfn for=Observable method><code>forEach(|callback|, |options|)</code></dfn> method steps are:
 
-    1. <span class=XXX>TODO: Spec this and use |callback| and |options|.</span>
+    1. Let |p| [=a new promise=].
```

Comment:
This seems generally useful but I'm curious if there was any discussion weighing up why this should return a promise, vs undefined per `Array#forEach`, or if this should perhaps be renamed to differentiate itself from any symmetry of the array equivalent?

Reply:
I think the analogy is that this kind of since array iteration is synchronous and returns `undefined`, therefore Observable "iteration" (i.e., `forEach()`) would return a _`Promise<undefined>`_, since Observable "iteration" is asynchronous, and it still could be useful to know when said iteration/visiting is "done".

There is some discussion about this on https://github.com/WICG/observable/issues/69, but basically I'm not sure there is much utility in any operator returning undefined ever, vs _`Promise<undefined>`_ which carries strictly more information. But I'd be open to re-consideration if you want to strike up more conversation or file another issue.

Reply:
I don't want to rock the boat on what is likely well-thought out and a culmination of very long discussions, but having said that, part of me wonders if this is a little awkward and maybe it's just... not worth implementing `forEach` at all? I wonder how much value it adds. Presumably given how much of this is modeled on existing user-land libraries we can find some kind of usage figures to guide us on which API should be prioritized? Is `forEach` high in that list?

Reply:
I'm sure you have an answer for the above so I filed https://github.com/WICG/observable/issues/106 to hopefully document that!